torch/_dynamo/output_graph.py

```diff
@@ -2229,6 +2198,274 @@ def maybe_lift_tracked_freevar_to_input(self, arg):
+        for i, s in enumerate(t.size()):
+            self._lift_symbols_in_symint(
+                s,
+                (
+                    TensorPropertySource(src, TensorProperty.SIZE, i)
+                    if src is not None
+                    else None
+                ),
+            )
+        if t.layout is torch.strided:
+            for i, s in enumerate(t.stride()):
+                self._lift_symbols_in_symint(
+                    s,
+                    (
+                        TensorPropertySource(src, TensorProperty.STRIDE, i)
+                        if src is not None
+                        else None
+                    ),
+                )
+            self._lift_symbols_in_symint(
+                t.storage_offset(),
+                (
+                    TensorPropertySource(src, TensorProperty.STORAGE_OFFSET)
+                    if src is not None
+                    else None
+                ),
+            )
+        elif t.layout is torch.sparse_coo:
+            self._lift_symbols_in_tensor(t._indices(), src)
+            self._lift_symbols_in_tensor(t._values(), src)
+        elif t.layout in {torch.sparse_csr, torch.sparse_bsr}:
+            self._lift_symbols_in_tensor(t.crow_indices(), src)
+            self._lift_symbols_in_tensor(t.col_indices(), src)
+        elif t.layout in {torch.sparse_csc, torch.sparse_bsc}:
+            self._lift_symbols_in_tensor(t.ccol_indices(), src)
+            self._lift_symbols_in_tensor(t.row_indices(), src)
+        if is_traceable_wrapper_subclass(t):
+            attrs, ctx = t.__tensor_flatten__()
+            for attr in attrs:
+                inner_t = getattr(t, attr)
+                self._lift_symbols_in_tensor(
+                    inner_t, AttrSource(src, attr) if src is not None else None
+                )
+
+    # Lookup the proxy in current tracer for each symbol in expressions of s,
+    # See Note [Auto lift basic free symbols when create_graph_input]
+    def lookup_bounded_symbols(
+        self, s: torch.SymInt
+    ) -> Tuple[Dict[sympy.Symbol, torch.fx.Proxy], Set[sympy.Symbol]]:
+        free_symbols = s.node.expr.free_symbols
+        if len(free_symbols) == 0:
+            return {}, set()
+
+        bounded = {}
+        to_be_bound = set()
+        for s0 in free_symbols:
+            if s0 not in self.bound_symbols:
+                to_be_bound.add(s0)
+                continue
+
+            proxy = self.bound_symbols[s0]
+            proxy = proxy() if isinstance(proxy, SubgraphTracer.LazyProxy) else proxy
+            assert (
+                proxy.tracer is self
+            ), f"The proxy of symbol {s0} doesn't belong to current tracer."
+            self.bound_symbols[s0] = proxy
+            bounded[s0] = proxy
+        return bounded, to_be_bound
+
+    # See NOTE: [Auto lift basic free symbols when create_graph_input]
+    def _lift_symbols_in_symint(
+        self, s: Union[int, torch.SymInt], source: Optional[Source]
+    ):
```

Comment:
I'm finding it difficult to follow the code. I don't have all of the context so I don't have a suggestion yet, but my thought process right now is:
- "_lift_symbols_in_symint" sounds like it will take all the symbols in the symint and make them inputs to the current subgraph tracer's graph.
- What is it returning? Sometimes it's None, sometimes it's... the bounded symbols?
- Why can't `_lift_symbols_in_symint` just always return None? Why do we need the return?